### ISSUE TITLE
Broken examples in the examples directory

### ISSUE THREAD

**Issue by havanagrawal** [thouska/spotpy#99]
`example_progressbar` doesn't work

```
python example_progressbar.py 
Traceback (most recent call last):
  File "example_progressbar.py", line 1, in <module>
    import spotpy.tools.progressbar as pb
ModuleNotFoundError: No module named 'spotpy.tools'
```

This is mostly because the `tools` directory has not been committed to the repo.

`tutorial_ownDatabase.py` is also broken.

```
python tutorial_ownDatabase.py 
Traceback (most recent call last):
  File "tutorial_ownDatabase.py", line 47, in <module>
    spot_setup=spot_setup()
  File "tutorial_ownDatabase.py", line 23, in __init__
    self.database = file('MyOwnDatabase.txt','w')
NameError: name 'file' is not defined
```

*Version Info:*

```
which python
/Users/havan/anaconda3/bin/python

python --version
Python 3.6.4 :: Anaconda custom (64-bit)

cat spotpy.egg-info/PKG-INFO 
Metadata-Version: 1.1
Name: spotpy
Version: 1.3.20
```

**Comment by thouska** [comment]
Thank you for this reported. The first issue (example_progressbar.py) did not find its way into spotpy so far. I will delete this file in the next upload.

Regarding to the tutorial_own_database.py: I hoped it would be sufficient to change `file('MyOwnDatabase.txt','w')` to `open('MyOwnDatabase.txt','w')`. However, then other stuff happens:

>File "_algorithm.py", line 269, in save
>   if like[0]>self.save_threshold: #Compares list/float
>IndexError: invalid index to scalar variable.

I am afraid a custom made database is not supported at the moment in spotpy. The build-in databases are initalized before they begin to save something. This errors with the given example. We would need to move the database initialization from `_algorithm.py` into `database.py`.

I think we need to have a closer look at this. Any help is warmly welcome.

**Comment by thouska** [comment]
Solved with #108
